dragonfly_doe2/doe/floor_space.py

```diff
@@ -88,25 +180,39 @@ def from_room(cls, room: Room2D):
                 'Expected dragonfly.room2d.Room2D'
             )
 
-        wall_constr_name = room.properties.energy.construction_set.wall_set.exterior_construction.display_name
+        wall_constr_name = short_name(
+            room.properties.energy.construction_set.wall_set.exterior_construction.display_name, 30)
         name = room.display_name
         walls = [
             Wall.from_room_seg(name, i, wall_constr_name)
             for i, bc in enumerate(room.boundary_conditions)
             if str(bc) == 'Outdoors'
         ]
+        slab = None
+        if room.is_ground_contact == True:
+            slab = Slab.from_checked_room(name, short_name(
+                room.properties.energy.construction_set.floor_set.ground_construction.display_name, 30))
+        roof = None
+        if room.is_top_exposed == True:
+            roof = RoofCeiling.from_room(room)
```

Comment:
In case you want you can write this one and the one for slab as a single line each.

```suggestion
        roof = None if room.is_top_exposed ! True else RoofCeiling.from_room(room)
```

Reply:
AHH!! I was drawing a blank on doing that in one line for some reason!